frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+
+import { ClusterVersionModel } from '../../models';
+import { ClusterVersionKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import {
+  CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL,
+  CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER,
+} from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [selectedRadio, setSelectedRadio] = React.useState(
+      currentUpstream ? 'custom' : 'default',
+    );
+    const [customUrl, setCustomUrl] = React.useState(currentUpstream);
+    const [invalidCustomUrl, setInvalidCustomUrl] = React.useState(false);
+
+    const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+      e.preventDefault();
+      if (selectedRadio === 'custom') {
+        if (!customUrl) {
```

Comment:
In other places in the admin perspective, we just make the input `required` and let the browser handle showing an error. This isn't exactly what PatternFly recommends, but would be consistent with other forms.

Reply:
@spadgett I tried setting the TextInput to `required` however it was not preventing form submission... Not sure why